flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/lookup/LookupOptions.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.connector.source.lookup;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.table.connector.source.lookup.cache.trigger.CacheReloadTrigger;
+import org.apache.flink.table.connector.source.lookup.cache.trigger.PeriodicCacheReloadTrigger;
+
+import java.time.Duration;
+
+/** Predefined options for lookup join. */
+public class LookupOptions {
+    public static final ConfigOption<LookupCacheType> CACHE_TYPE =
+            ConfigOptions.key("lookup.cache")
+                    .enumType(LookupCacheType.class)
+                    .defaultValue(LookupCacheType.NONE)
+                    .withDescription("The caching strategy for this lookup table");
+
+    public static final ConfigOption<ReloadStrategy> FULL_CACHE_RELOAD_STRATEGY =
+            ConfigOptions.key("lookup.full-cache.reload-strategy")
+                    .enumType(ReloadStrategy.class)
+                    .defaultValue(ReloadStrategy.PERIODIC)
+                    .withDescription(
+                            "Defines which strategy to use to reload full cache: "
+                                    + "PERIODIC - cache is reloaded with fixed intervals without initial delay; "
+                                    + "TIMED - cache is reloaded at specified time with fixed intervals multiples of one day.");
+
+    public static final ConfigOption<Duration> FULL_CACHE_PERIODIC_RELOAD_INTERVAL =
+            ConfigOptions.key("lookup.full-cache.periodic-reload.interval")
+                    .durationType()
+                    .defaultValue(Duration.ZERO)
```

Comment:
I'm not sure if it is a good idea to have a default value with 0 here. If users intend to reload the cache periodically, the interval should be set in the configuration to fully express the reloading strategy. Maybe we could just leave it without default value here and have a check in the `PeriodicCacheReloadTrigger`. WDYT?

Reply:
Yes, I agree. If partial cache can't be built without specifying options for it, the same should be for full cache.